api/src/paths/taxonomy/taxon/tsn/hierarchy.ts

```diff
@@ -0,0 +1,106 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { getAPIUserDBConnection } from '../../../../database/db';
+import { ItisService } from '../../../../services/itis-service';
+import { getLogger } from '../../../../utils/logger';
+
+const defaultLog = getLogger('paths/taxonomy/taxon/tsn/hierarchy');
+
+export const GET: Operation = [getTaxonHierarchyByTSN()];
+
+GET.apiDoc = {
+  description: 'Get taxon records by TSN ids.',
+  tags: ['taxon_id'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      description: 'Taxon TSN ids.',
+      in: 'query',
+      name: 'tsn',
+      schema: {
+        type: 'array',
+        description: 'One or more Taxon TSN ids.',
+        items: {
+          type: 'integer',
+          minimum: 0,
+          minItems: 1,
+          maxItems: 100
+        }
+      },
+      required: true
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Taxonomy response.',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'array',
+            items: {
+              title: 'Species',
+              type: 'object',
+              required: ['tsn', 'hierarchy'],
+              properties: {
+                tsn: {
+                  type: 'integer'
+                },
+                hierarchy: { type: 'array', items: { type: 'integer' } }
```

Comment:
Small detail, but are the TSNs in the array in some order? kingdom -> target_tsn? Or something like that?

A description explaining that might be valuable

Reply:
Fixed